src/multihash.rs

```diff
@@ -1,121 +1,99 @@
-use crate::error::Error;
-use crate::hasher::Digest;
+use crate::hasher::{Digest, Size};
+use crate::Error;
+use core::convert::TryFrom;
 #[cfg(feature = "std")]
 use core::convert::TryInto;
 use core::fmt::Debug;
+use generic_array::GenericArray;
 
-/// Trait for reading and writhing Multihashes.
+/// Trait that implements hashing.
 ///
-/// It is usually derived from a list of hashers by the [`Multihash` derive].
+/// It is usually implemented by a custom code table enum that derives the [`Multihash` derive].
 ///
 /// [`Multihash` derive]: crate::derive
-pub trait MultihashDigest: Clone + Debug + Eq + Send + Sync + 'static {
-    /// Returns the hash of the input.
-    fn new(code: u64, input: &[u8]) -> Result<Self, Error>;
+pub trait MultihashCode: TryFrom<u64> + Into<u64> + Send + Sync + 'static {
```

Comment:
Can you make it `Unpin`, `Copy`, `Eq` and `Debug`? Those can be easily derived for enums and make life easier.